src/components/views/settings/tabs/user/GeneralUserSettingsTab.tsx

```diff
@@ -238,10 +244,20 @@ export default class GeneralUserSettingsTab extends React.Component<IProps, ISta
 
     private onSpellCheckLanguagesChange = (languages: string[]): void => {
         this.setState({ spellCheckLanguages: languages });
+        PlatformPeg.get()?.setSpellCheckLanguages(languages);
+    };
 
-        const plaf = PlatformPeg.get();
-        if (plaf) {
-            plaf.setSpellCheckLanguages(languages);
+    private onSpellCheckEnabledChange = (spellCheckEnabled: boolean): void => {
+        this.setState({ spellCheckEnabled });
+
+        if (spellCheckEnabled && IS_MAC) {
+            // XXX: On macOS the languages are controlled by the system
+            // settings, so we just set something, so that spell check gets
+            // enabled
+            PlatformPeg.get()?.setSpellCheckLanguages(["en_US"]);
```

Comment:
we should probably just turn it onto a simple toggle if we can't support the platform.

Reply:
Currently, the intention is for no spell check languages to be the same as spell check being disabled, so it makes sense to only have a method that sets the spell check languages. 

I'd like to wait to see what Jano thinks of the current behaviour and make changes then, if necessary